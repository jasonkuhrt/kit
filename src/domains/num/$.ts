/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
