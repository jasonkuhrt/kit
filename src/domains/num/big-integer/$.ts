export * as BigInteger from './$$.ts'
