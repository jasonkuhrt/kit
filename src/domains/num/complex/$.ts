export * as Complex from './$$.js'
