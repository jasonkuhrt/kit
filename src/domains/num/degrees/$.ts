// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
