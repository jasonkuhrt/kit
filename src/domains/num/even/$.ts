// Export the namespace containing all Even operations
export * as Even from './$$.ts'
