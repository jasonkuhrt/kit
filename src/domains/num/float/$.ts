// Export the namespace containing all Float operations
export * as Float from './$$.js'
