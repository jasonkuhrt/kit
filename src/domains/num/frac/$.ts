// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
