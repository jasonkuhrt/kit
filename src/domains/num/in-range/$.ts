// Export the namespace containing all InRange operations
export * as InRange from './$$.js'
