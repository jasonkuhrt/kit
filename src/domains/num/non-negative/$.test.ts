import { expect, test } from 'vitest'
import { NonNegative } from './$.js'

test('isNonNegative', () => {
  // Valid cases - non-negative numbers (>= 0)
  expect(NonNegative.is(0)).toBe(true)
  expect(NonNegative.is(1)).toBe(true)
  expect(NonNegative.is(100)).toBe(true)
  expect(NonNegative.is(0.5)).toBe(true)
  expect(NonNegative.is(Number.MAX_VALUE)).toBe(true)
  expect(NonNegative.is(Infinity)).toBe(true)

  // Invalid cases - negative numbers
  expect(NonNegative.is(-1)).toBe(false)
  expect(NonNegative.is(-0.1)).toBe(false)
  expect(NonNegative.is(-100)).toBe(false)
  expect(NonNegative.is(-Infinity)).toBe(false)
  expect(NonNegative.is(NaN)).toBe(false)
  expect(NonNegative.is('0')).toBe(false)
  expect(NonNegative.is(null)).toBe(false)
})

test('tryNonNegative', () => {
  // Valid cases
  expect(NonNegative.tryFrom(0)).toBe(0)
  expect(NonNegative.tryFrom(1)).toBe(1)
  expect(NonNegative.tryFrom(100)).toBe(100)
  expect(NonNegative.tryFrom(0.5)).toBe(0.5)
  expect(NonNegative.tryFrom(Number.MAX_VALUE)).toBe(Number.MAX_VALUE)
  expect(NonNegative.tryFrom(Infinity)).toBe(Infinity)

  // Invalid cases - should return null
  expect(NonNegative.tryFrom(-1)).toBe(null)
  expect(NonNegative.tryFrom(-0.1)).toBe(null)
  expect(NonNegative.tryFrom(-100)).toBe(null)
  expect(NonNegative.tryFrom(-Infinity)).toBe(null)
  expect(NonNegative.tryFrom(NaN)).toBe(null)
})
