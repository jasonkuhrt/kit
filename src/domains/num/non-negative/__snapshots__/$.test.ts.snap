// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`is > is("0") > is("0") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
"0"
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.1) > is(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-100) > is(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.5) > is(0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.7976931348623157e+308) > is(1.7976931348623157e+308) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.7976931348623157e+308
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.1) > tryFrom(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-100) > tryFrom(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.5) > tryFrom(0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
0.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.7976931348623157e+308) > tryFrom(1.7976931348623157e+308) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.7976931348623157e+308
╠══════════════════════════════════════════════════╣ THEN RETURNS
1.7976931348623157e+308
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(100) > tryFrom(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS
100
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;
